cmd/cat/cat.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+)
+
+func checkError(err error) {
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func printFileContents(reader io.Reader, n bool) {
+
+	scanner := bufio.NewScanner(reader)
+	lineNumber := 1
+
+	for scanner.Scan() {
+		if n {
+			fmt.Printf("    %d ", lineNumber)
+		}
+		fmt.Println(scanner.Text())
+		lineNumber++
+	}
+
+}
+func main() {
+
+	var n bool
+	flag.BoolVar(&n, "n", false, "Numbers output lines")
+	flag.Parse()
+
+	if len(flag.Args()) == 0 {
+		printFileContents(os.Stdin, n)
+		return
+	}
+
+	for _, arg := range flag.Args() {
+		file, err := os.Open(arg)
+		checkError(err)
+
+		printFileContents(file, n)
+		file.Close()
```

Comment:
you should use [defer](https://gobyexample.com/defer)
```suggestion
		defer file.Close()
```